### PR TITLE
BFD-2369: Diagnosis Type Page Resolution

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-2369](https://jira.cms.gov/browse/BFD-2369)

**User Story or Bug Summary:**

As a BFD user, I want https://bluebutton.cms.gov/resources/codesystem/diagnosis-type to resolve so that I can learn more about the diagnosis type coding system.

### What Does This PR Do?

This is one of several tickets involved with BFD's ability to serve URLs that resolve pertaining to the blue button static site. 
In this ticket, the focus is the diagnosis type page.

A markdown file was added so that a new page can be autogenerated and deployed.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check these things, in particular:

* Does this page conform to the content standards and adequately describe the resource in question?

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then here's a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence: N/A.
    * Does this PR add any new software dependencies? **No**.
    * Does this PR modify or invalidate any of our security controls? **No**.
    * Does this PR store or transmit data that was not stored or transmitted before? **No**.
* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons? **No**.

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following pre-requisite changes have been fully deployed:

* CMSgov/some_repo#42

### Submitter Checklist

I have gone through and verified that...:

* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] I have named this PR and its branch such that they'll be automatically be linked to the (most) relevant Jira issue, per: <https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html>.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: _pages/resources-codesystem-diagnosis-type.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+permalink: '/resources/codesystem/diagnosis-type/'
+---
+# Coding System: Diagnosis Type
+
+## Usage
+
+**System URI**: `https://bluebutton.cms.gov/resources/codesystem/diagnosis-type`
+
+## Description
+This variable describes the type of diagnosis for a patient. In a setting such as a hospital, several events take place in diagnosing a patient that depend on whether a patient is admitted or discharged, if there is more than one diagnosis being recorded, or one diagnosis is more important than another.
```
Comment (williamharty) [inline _pages/resources-codesystem-diagnosis-type.md:13]: I would suggest just saying the following:
This variable provides the diagnosis type for the corresponding diagnosis code.

File: _pages/resources-codesystem-diagnosis-type.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+**Values for Codeset**:
+
+| Value     | Description                                                         |
+|-----------|---------------------------------------------------------------------|
+| admitting | A patient is admitted to a hospital.                                |
+| primary   | The most important or first diagnosis recorded upon admission.      |
+| secondary | A less important or not the first diagnosis recorded upon admission.|
+| discharge | A patient is discharged from a hospital.                            |
+
```
Comment (williamharty) [inline _pages/resources-codesystem-diagnosis-type.md:26]: As far as I can tell, this codesystem was only used in V1.  The DD shows different possible codes.  It's quite possible the DD is outdated or simply wrong, as the list you have above makes more sense.

Here's what the DD has:

- admitting
- external-first
- principal
- external
- reason-for-visit

Comment (williamharty) [inline _pages/resources-codesystem-diagnosis-type.md:26]: Here is some suggested descriptions for each of the above codes

| Value     | Description                                                         |
|-----------|---------------------------------------------------------------------|
| admitting | A diagnosis code indicating the beneficiary's initial diagnosis at admission   |
| primary   | A diagnosis code identifying the diagnosis, condition, problem, or other reason for the admission/encounter/visit shown in the medical record to be chiefly responsible for the services provided.      |
| secondary | A diagnosis other than the primary diagnosis |
| discharge | The diagnosis given when the patient is discharged from the hospital  |

**Review by gmaciolek — APPROVED** [review]
I agree with Bill's suggestions and recommendations.

**Review by williamharty — COMMENTED** [review]
see inline comments

File: _pages/resources-codesystem-diagnosis-type.md (hunk 2 above)
Comment (JeremyIR) [reply _pages/resources-codesystem-diagnosis-type.md:26]: Interesting. Yeah I found the diagnosis types I provided within the bluebutton static site. Perhaps we can just omit discharge. We can always amend the page if we have more certainty on these

File: _pages/resources-codesystem-diagnosis-type.md (hunk 1 above)
Comment (JeremyIR) [reply _pages/resources-codesystem-diagnosis-type.md]: Done

**Review by williamharty — APPROVED** [review]

**Pull request merged**